src/nifake/tests/test_session.py

```diff
@@ -1,12 +1,72 @@
-import math
+import ctypes
+# import math
 import mock_helper
 import nifake
-import sys
-import warnings
+# import sys
+from nifake import visatype
+# import warnings
 
-from mock import ANY
+# from mock import ANY
 from mock import patch
 
+
+# Matching classes for setting mock expectations:
+
+class ScalarMatcher(object):
+    def __init__(self, expected_type, expected_value):
+        self.expected_type = expected_type
+        self.expected_value = expected_value
+
+    def __repr__(self):
+        return 'Matcher({0}, {1})'.format(self.expected_type, self.expected_value)
+
+    def __eq__(self, other):
+        if not isinstance(other, self.expected_type):
+            print("Unexpected type. Expected: {0}. Received: {1}".format(self.expected_type, type(other)))
+            return False
+        if other.value == self.expected_value:
+            return True
+        else:
+            print("Unexpected value. Expected: {0}. Received: {1}".format(self.expected_value, other.value))
+        return
+
+
+class StringMatcher(object):
+    def __init__(self, expected_string_value):
+        self.expected_string_value = expected_string_value
+
+    def __eq__(self, other):
```

Comment:
We know it is a string, should you also check that the element type is ViChar?

Reply:
Checking that element is of type byte.